### PR TITLE
Adding S-parameter definition in Touchstone V1 comment

### PR DESCRIPTION
**Pull request by Asachoo**
### **Description of the PR**
Try to fix the issue https://github.com/scikit-rf/scikit-rf/issues/1116.

In this PR, the reference impedance convention (`s_def`) will be included in the TouchStone file if `write_z0=True` is set in the  `write_touchstone` method.

### **Touchstone File saving change**
For example, the comments in the `skrf/tests/tmp_skrf_oneport_powerwave.s1p` file created by `scikit-rf` are updated from:
```
!Created with skrf (http://scikit-rf.org).
!Data is not renormalize
```
to:
```
! Created with skrf (http://scikit-rf.org).
! Data is not renormalized according to the power convention.
```
The keywords `power convention`, `pseudo convention` and `traveling convention` are used in comments to identify the reference impedance convention (`s_def`).

### **Touchstone File loading change**
If the reference impedance convention (`s_def`) is specified in the comments, that convention will be used. Otherwise, the default `S_DEF_HFSS_DEFAULT` (i.e. the `traveling` definition) will be applied in `hfss_impedance` pattern.

### CODE REVIEW

**Comment by Asachoo** [comment]
The output of the case in https://github.com/scikit-rf/scikit-rf/issues/1116 becomes:
```bash
skrv version: 1.1.0

buffer's s_def: power
traveling_file's s_def: power
power_file's s_def: power

Saved renormalized Network (traveling) is equal to buffer? True
Saved renormalized Network (power) is equal to buffer? True

Renormalize back to 50 Ohm Network is equal (buffer vs traveling)? True
Renormalize back to 50 Ohm Network is equal (buffer vs power)? True
```
This demonstrates that the reference impedance convention (`s_def`) can be obtained and the `S-parameters` can be correctly recovered.

File: skrf/tests/tmp_skrf_oneport_powerwave.s1p [hunk 1]
```diff
@@ -8,8 +8,8 @@
 ! YY = 1
 !Data is not renormalized
 !
-!Created with skrf (http://scikit-rf.org).
-!Data is not renormalized
+! Created with skrf (http://scikit-rf.org).
+! Data is not renormalized according to the power convention.
```
Comment (jhillairet) [inline skrf/tests/tmp_skrf_oneport_powerwave.s1p:12]: I find this sentence confusing. How about splitting into two sentences like:

```
! Data is not renormalized
! S-parameter uses the "{}" definition 
```
?

Comment (Asachoo) [reply skrf/tests/tmp_skrf_oneport_powerwave.s1p]: Changes made as suggested.

**Review by jhillairet — APPROVED** [review]
Any comment from others ? @Ttl, @mhuser @biergaizi @FranzForstmayr  ?

**Review by mhuser — APPROVED** [review]
LGTM

**Review by FranzForstmayr — APPROVED** [review]

**Pull request merged**